### PR TITLE
Twitter timeline hidden, unless loaded

### PR DESCRIPTION
**Pull request by digital-codes**
Twitter timeline on Blog hidden by default, to suppress anoying (almost) empty tile on Firefox strict privacy.  Tile shown after successful loading. Introduces a small delay, could be shortend, maybe.

### CODE REVIEW

File: themes/codefor-theme/layouts/blog/list.html [hunk 1]
```diff
@@ -13,8 +13,9 @@
                 {{ partial "blog-preview.html" . }}
             {{ end }}
 
+            <!-- For Twitter we add the hidden class which will be removed in script below -->
             {{if eq $paginator.PageNumber 1}}
-            <div class="col-24 col-md-12 mb-5 mb-md-7 project-preview-container twitter-container" >
+            <div class="col-24 col-md-12 mb-5 mb-md-7 project-preview-container twitter-container hidden" >
```
Comment (ubergesundheit) [inline themes/codefor-theme/layouts/blog/list.html:18]: Bootstrap already contains such a helper class (`d-none`)

```suggestion
            <div class="col-24 col-md-12 mb-5 mb-md-7 project-preview-container twitter-container d-none" >
```

File: themes/codefor-theme/layouts/blog/list.html [hunk 2]
```diff
@@ -37,6 +38,8 @@
     var totalPages = $('#blog-entries-container').data('total-pages');
     var currentpage = 1;
     var loading = false;
+    setTimeout(checkTwitter,3000); // check for twitter
```
Comment (ubergesundheit) [inline themes/codefor-theme/layouts/blog/list.html:41]:
```suggestion
    setTimeout(checkTwitter, 3000); // execute twitter widget check after a delay
```

File: themes/codefor-theme/layouts/blog/list.html [hunk 3]
```diff
@@ -54,6 +57,20 @@
...
+            // show container only if this happens ...
+			var f = document.getElementById("twitter-widget-0");
+			f.parentNode.classList.remove("hidden");
+			} catch {
+				console.log("No twitter widget");
+			}
+			//setTimeout(checkTwitter,1000)
+	}
+
+
```
Comment (ubergesundheit) [inline themes/codefor-theme/layouts/blog/list.html:73]: IMO we should avoid the unnecessary try-catch block and `console.log` call.

```suggestion
function checkTwitter() {
    var widgetNode = document.querySelector("#twitter-widget-0");
    if (widgetNode) {
        widgetNode.parentNode.classList.remove("d-none");
    }
}
```

File: themes/codefor-theme/assets/scss/main.scss [hunk 4]
```diff
@@ -259,3 +259,7 @@ blockquote {
     margin-bottom: 15px;
 }
 
+// hide element, e.g. twitter timeline
+.hidden {
+  display:none;
+}
```
Comment (ubergesundheit) [inline themes/codefor-theme/assets/scss/main.scss:265]: Please remove this, bootstrap already contains a helper class to set the `display` style property of elements to `none`. (`d-none`)

**Review by ubergesundheit — CHANGES_REQUESTED** [review]

**Review by digital-codes — COMMENTED** [review]
Changes from Gerald applied. Thanks

**Review by ubergesundheit — APPROVED** [review]
Nice 👍

**Pull request merged**